systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -358,6 +363,51 @@ public static String getImageValueFromCO(String name) {
         }
         return "";
     }
+    /**
+     * Log actual status of custom resource with pods.
+     * @param customResource - Kafka, KafkaConnect etc. - every resource that HasMetadata and HasStatus (Strimzi status)
+     */
+    public static <T extends HasMetadata & HasStatus> void logCurrentStatus(T customResource) {
+        String kind = customResource.getKind();
+        String name = customResource.getMetadata().getName();
+
+        List<String> log = new ArrayList<>(asList("\n", kind, " status:\n", "\nConditions:\n"));
+
+        for (Condition condition : customResource.getStatus().getConditions()) {
+            log.add("\tType: " + condition.getType() + "\n");
+            log.add("\tMessage: " + condition.getMessage() + "\n");
+        }
+
+        PodUtils.logCurrentPodStatus(name, log);
+    }
+
+    /**
+     * Wait until the CR is in desired state
+     * @param operation - client of CR - for example kafkaClient()
+     * @param resource - custom resource
+     * @param status - desired status
+     * @return returns CR
+     */
+    public static <T extends HasMetadata & HasStatus> T waitForStatus(MixedOperation<T, ?, ?, ?> operation, T resource, String status) {
+        List<String> printWholeCr = new ArrayList<>(asList("KafkaConnector", "KafkaTopic", "KafkaUser"));
+
+        TestUtils.waitFor("Wait for {}" + resource.getKind() + ":{} " + resource.getMetadata().getName() + "will have desired state {}" + status,
+            Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, Constants.TIMEOUT_FOR_RESOURCE_READINESS,
+            () -> operation.inNamespace(resource.getMetadata().getNamespace())
+            .withName(resource.getMetadata().getName())
+            .get().getStatus().getConditions().stream().anyMatch(condition -> condition.getType().equals(status)),
```

Comment:
I tried whole `KafkaST` (I know we tried to fix the problem when `WARNING` appear to condition) and it should work. I will take a deeper look at it and let you know if it will cause any future problems.